test/bot.coffee

```diff
@@ -1,27 +1,32 @@
 should = require 'should'
 chai = require 'chai'
-{ EnterMessage, LeaveMessage, TopicMessage, CatchAllMessage, Robot } = require.main.require 'hubot'
+mockery = require 'mockery'
+hubotSlackMock = require '../slack.coffee'
+mockery.registerMock('hubot-slack', hubotSlackMock);
+
+{ EnterMessage, LeaveMessage, TopicMessage, CatchAllMessage, Robot, loadBot } = require.main.require 'hubot'
 { SlackTextMessage, ReactionMessage, PresenceMessage } = require '../src/message'
 SlackClient = require '../src/client'
 _ = require 'lodash'
 
 describe 'Adapter', ->
+  before ->
+    mockery.enable({
+      warnOnUnregistered: false
```

Comment:
out of curiosity, why is this set to `false`?

Reply:
Unregistered means all of the `requires` we aren't mocking, so every package used in the tests  prints a warning. This makes it so that won't happen.